### ISSUE TITLE
Large quantity of facets causing degradation of Meilisearch performance

### ISSUE THREAD

**Issue by shivaylamba** [meilisearch/meilisearch#2349]
**Describe the bug**
I am building an [e-commerce demo](https://medusajs-storefront.vercel.app/) using an Amazon Dataset that contains more than 1 million items. It has 300k brands, 100k tags which are added as part of filterable attributes. I am using Instant Meilisearch and thus these are also part of the Refinement list. 
The time required to calculate the facets will increase as the number of facets increases, hence degrading search performance. Since there are 300K brands and 100K tags, it can take a lot of time to load them initially. 

**To Reproduce**
1. Visit https://medusajs-storefront.vercel.app/
2. Click on Inspect element to open Developer tools 
3. Go to the networks section and click on the search request
4. Click on the Response tab
5. You can see the response. 

**Expected behavior**
Performance should be quick. 

**Screenshots**
![image](https://user-images.githubusercontent.com/19529592/165126748-6544f9c2-ee0f-44d0-91dd-91626bb8b63c.png)

![image](https://user-images.githubusercontent.com/19529592/165150095-76857b8d-87ad-4f4d-97ee-9dbfd5f0e32e.png)

**Meilisearch version:** v0.26.0 

**Additional context**
Additional information that may be relevant to the issue.
[e.g. architecture, device, OS, browser]
Browser: Chrome
OS: MacOS 
Device: Mac Book Pro

**Comment by Kerollmops** [comment]
Hey @shivaylamba,

I ran the query that was taking a lot of time and it is indeed related to the number of facets returned in the results. More specifically we are returning 172700 facet values associated with their counts, which is a lot.

You can run this `jq` query on the file I linked, here. Make sure to unzip it first.

[output.json.zip](https://github.com/meilisearch/meilisearch/files/8556325/output.json.zip)

```
cat output.json | jq '. | .facetsDistribution.brand | length'
```

**Comment by shivaylamba** [comment]
Thanks for your response! 

So for now would you recommend reducing the number of facets value @Kerollmops

**Comment by curquiza** [comment]
Discussed with @gmourier and @Kerollmops -> this will be fixed in v0.28.0 by introducing an hard limit (will be customizable in the future, but not for v0.28.0)
I will open an issue regarding this next week

**Comment by shivaylamba** [comment]
Alright thank you @curquiza
